src/components/SendModal/index.tsx

```diff
@@ -25,56 +22,16 @@ interface SendModalProps {
 }
 
 const SendModal = ({ isOpen, onClose }: SendModalProps) => {
-  const toast = useToast();
-  const router = useRouter();
   const searchParams = useSearchParams();
-  const { selectedAnswer } = useContext(AppContext);
   const [loading, setLoading] = useState(false);
   const [responderName, setResponderName] = useState('');
+  const sendEmail = useSendEmail();
 
   const encodedSenderName = searchParams.get('sn')!;
   const senderName = decodeText(encodedSenderName);
-  const senderEmail = searchParams.get('se')!;
 
-  const handleClick = async () => {
-    if (!selectedAnswer)
-      return toast({
-        title: `Oops! Something went wrong (╥﹏╥)`,
-        status: 'error',
-        duration: 5000,
-        isClosable: true,
-      });
-
-    setLoading(true);
-
-    await emailService({
-      senderEmail,
-      selectedAnswer,
-      responderName,
-      senderName,
-    })
-      .then(() => {
-        setLoading(false);
-        onClose();
-        router.push('/');
-
-        return toast({
-          title: `Thanks! We'll let ${senderName} know`,
-          status: 'success',
-          duration: 5000,
-          isClosable: true,
-        });
-      })
-      .catch(() => {
-        setLoading(false);
-        return toast({
-          title: `Oops! Something went wrong (╥﹏╥)`,
-          status: 'error',
-          duration: 5000,
-          isClosable: true,
-        });
-      });
-  };
+  const handleClick = async () =>
```

Comment:
No need for indirection into a handler function here. It's a one-liner, just make the sendEmail function call directly in your prop or wherever handleClick is passed or called. This will communicate intent more clearly.

Reply:
After comments below, you may want indirection again, but if it stays a one-liner, there's no need for handleClick.